src/ui/CodeRenderer/CodeRenderer.spec.tsx

```diff
@@ -0,0 +1,171 @@
+          cb()
+        }, 50)
+        return 1
+      })
+
+      render(
+        <CodeRenderer
+          code={code}
+          fileName="sample.py"
+          rendererType="SINGLE_LINE"
+          LineComponent={({ i }) => (
+            <tr key={i}>
+              <td>this is a random line test component {i + 1}</td>
+            </tr>
+          )}
+        />
+      )
+
+      const lines = await screen.findAllByText(
+        /this is a random line test component/
+      )
+      expect(lines[0]).toBeInTheDocument()
+
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      const table = screen.getByRole('table')
+      await waitFor(() => expect(table).toHaveStyle('pointer-events: none'))
+    })
+
+    it('reset pointer events after scroll', async () => {
+      dateNowSpy
+        .mockImplementationOnce(() => 1000)
+        .mockImplementationOnce(() => 2000)
+      requestAnimationFrameSpy.mockImplementation((cb) => {
+        setTimeout(() => {
+          cb()
+        }, 50)
+        return 1
+      })
+
+      render(
+        <CodeRenderer
+          code={code}
+          fileName="sample.py"
+          rendererType="SINGLE_LINE"
+          LineComponent={({ i }) => (
+            <tr key={i}>
+              <td>this is a random line test component {i + 1}</td>
+            </tr>
+          )}
+        />
+      )
+
+      const lines = await screen.findAllByText(
+        /this is a random line test component/
+      )
+      expect(lines[0]).toBeInTheDocument()
+
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      const table = screen.getByRole('table')
+      await waitFor(() => expect(table).toHaveStyle('pointer-events: none'))
+      await waitFor(() => expect(table).toHaveStyle('pointer-events: auto'))
+    })
+
+    it('calls cancelAnimationFrame', async () => {
+      dateNowSpy
+        .mockImplementationOnce(() => 1000)
+        .mockImplementationOnce(() => 2000)
+      requestAnimationFrameSpy.mockImplementation((cb) => {
+        setTimeout(() => {
+          cb()
+        }, 50)
+        return 1
+      })
+
+      const { container } = render(
+        <CodeRenderer
+          code={code}
+          fileName="sample.py"
+          rendererType="SINGLE_LINE"
+          LineComponent={({ i }) => (
+            <tr key={i}>
+              <td>this is a random line test component {i + 1}</td>
+            </tr>
+          )}
+        />
+      )
+
+      const lines = await screen.findAllByText(
+        /this is a random line test component/
+      )
+      expect(lines[0]).toBeInTheDocument()
+
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      // eslint-disable-next-line testing-library/no-container
+      container.remove()
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      await waitFor(() => expect(cancelAnimationFrameSpy).toHaveBeenCalled())
```

Comment:
I do not understand what is going on in this test specifically - maybe we can go over this PR in our 1-1 on Tuesday?

Reply:
Yes of course!